compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaPathParameterSymbol.java

```diff
@@ -123,12 +140,13 @@ public String signature() {
 
         String typeSignature;
         if (this.pathSegmentKind() == Kind.PATH_REST_PARAMETER) {
-            typeSignature = ((ArrayTypeSymbol) this.typeDescriptor()).memberTypeDescriptor().signature() + "...";
+            typeSignature = this.typeDescriptor().signature() + "...";
         } else {
             typeSignature = this.typeDescriptor().signature();
         }
 
-        this.signature = "[" + typeSignature + " " + this.getName().get() + "]";
+        String paramName = this.getName().isPresent() ? " " + this.getName().get() : "";
+        this.signature = "[" + typeSignature + paramName + "]";
```

Comment:
I would combine this to avoid an unnecessary concatenation when there is no param name.

Reply:
Like this? 
```Java
this.signature = "[" + typeSignature + (this.getName().isPresent() ? " " + this.getName().get() : "") + "]";
```
I feel like this takes a little bit of effort to understand.

How about something like this?
```Java
this.getName().ifPresentOrElse(
                name -> this.signature = "[" + typeSignature + " " + name + "]",
                () -> this.signature = "[" + typeSignature + "]");
```

Reply:
I would just do

```java
this.signature = this.getName().isPresent() ? 
                "[" + typeSignature + " " + this.getName().get() + "]" :
                "[" + typeSignature + "]"
```

Reply:
Ack.